include/compartment.h

```diff
@@ -0,0 +1,105 @@
+#ifndef _COMPARTMENT_H
+#define _COMPARTMENT_H
+
+#include <assert.h>
+#include <elf.h>
+#include <fcntl.h>
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/auxv.h>
+#include <sys/mman.h>
+#include <sys/stat.h>
+#include <unistd.h>
+
+#include "cheriintrin.h"
+
+#include "manager.h"
+
+#define align_down(x, align)    __builtin_align_down(x, align)
+#define align_up(x, align)      __builtin_align_up(x, align)
+
+#define SEG_MAX_COUNT 20
+
+struct func_intercept;
+
+struct intercept_patch
+{
+    int* patch_addr;
+    int instr;
+};
+
+struct SegmentMap
```

Comment:
A brief doc comment here would be great. I also wasn't sure if all the members of the struct were required? I wonder if `mem_top-mem_bot==mem_sz`?

Reply:
I do need to add a lot of comments. I've added a placeholder for now, with a high-level definition in https://github.com/capablevms/CHERI-ELF-comp/pull/1/commits/78116572f132210f1599649c968d976f1076bd46. I can expand it if you think it needs expanding at this stage.

I would like to keep this redundancy for now, since it makes it easier to reason about some things, and that it also helps me more easily reference the Morello example I'm following. There are probably a few more struct members that can be removed, but I would like to do that once the codebase becomes more stable.

Reply:
I'm fine with the redundancy especially if we (at least briefly) document that there is redundancy.

Reply:
I've added a more generic to do to revisit struct members and definitions in https://github.com/capablevms/CHERI-ELF-comp/pull/1/commits/de7d74516420bf8274d87266ac07569e048f6e45.